dex/networks/eth/contracts/ETHSwapV0.sol

```diff
@@ -14,47 +42,75 @@ contract ETHSwap {
         State state;
     }
 
+    // Swaps is a map of swap secret hashes to swaps. It can be read by anyone
+    // for free.
     mapping(bytes32 => Swap) public swaps;
 
+    // constructor is empty. This contract has no connection to the original
+    // sender after deployed. It can only be interacted with by users
+    // initiating, redeeming, and refunding swaps.
     constructor() {}
 
-    modifier isRefundable(bytes32 secretHash, address refunder) {
+    // isRefundable checks that a swap can be refunded. The requirements are
+    // the initiator is msg.sender, the state is Filled, and the block
+    // timestamp be after the swap's stored refundBlockTimestamp.
+    modifier isRefundable(bytes32 secretHash) {
```

Comment:
The reasoning for removing was simply because it didn't seem necessary as we are just sending `msg.sender` anyway. So, for simplicity.

I have not used etherscan.io's read contract feature... I will go check it out. However, msg.send should be public as part of the initial transaction?

Reply:
e.g.  some random contract https://etherscan.io/address/0xb1e9157c2fdcc5a856c8da8b2d89b6c32b3c1229#readContract

You can do stuff like balanceOf(account) in the explorer without messing with transactions

Reply:
It looks like you can submit code to get some of that debugging. You can see that this contract is not set up to do that: https://etherscan.io/address/0x7f3a4085dcdccd5d52249dbf604ba9236b433040#code

Reply:
I think we could make some separate public view functions that inform about states, and so would be visible and useable by an explorer. I read somewhere that public functions shouldn't call other public functions. Probably because of vulnerabilities like reentry attack creeping in undetected. Will look into it some more, but separate public functions for debugging like you say should be doable and fine, if you think it would be good.

Reply:
I'm sure we'll want such methods, and the contract source will be published and probably submitted too, but it's not something to hold up this PR anyway.
I expect these separate public functions won't add to gas cost for regular swaps, just the tx that publishes the actual contract, so I figure it's alright to add any such functions to the contract interface, right?

Reply:
Yes, I think that the gas cost for the initial contract would be higher, probably not so much higher, but it shouldn't affect other methods at all.